corehq/apps/app_manager/static/app_manager/js/source_files.js

```diff
@@ -10,36 +10,44 @@ hqDefine('app_manager/js/source_files',[
             e.preventDefault();
             $(this).parents('tr').next('tr').toggleClass("hide");
         });
-    
+
         var currentVersion = initialPageData.get('current_version'),
-            builtVersions = initialPageData.get('built_versions'),
             $form = $("#compare-form"),
-            $input = $form.find("input");
+            $select = $form.find("select");
 
-        builtVersions = _.sortBy(_.filter(builtVersions, function (v) {
-            return v.version !== currentVersion;
-        }), function (v) { return parseInt(v.version); }).reverse();
-        var versionMap = _.indexBy(builtVersions, 'version');
-    
-        $input.select2({
-            data: _.map(builtVersions, function (v) {
-                return {
-                    id: v.version,
-                    text: v.version + ": " + (v.comment || "no comment"),
-                };
-            }),
+        $select.select2({
+            ajax: {
+                url: initialPageData.reverse('paginate_releases'),
+                dataType: 'json',
+                data: function (params) {
+                    return {
+                        limit: 10,
+                        query: params.term,
+                        page: params.page,
+                    };
+                },
+                processResults: function (data) {
+                    return {
+                        results: _.map(data.apps, function (build) {
+                            return {
+                                id: build.id,
+                                text: build.version + ": " + (build.build_comment || gettext("no comment")),
+                            };
+                        }),
+                        pagination: data.pagination,
+                    };
+                },
+            },
+            width: '200px',
         });
-    
+
         $form.find("button").click(function () {
-            var version = $input.val();
+            var version = $select.val();
             if (!version) {
-                alert("Please enter a version to compare");
-                return;
-            } else if (!versionMap[version]) {
-                alert(version + " is not a valid version");
+                alert(gettext("Please enter a version to compare"));
                 return;
             }
-            window.location = initialPageData.reverse('diff', versionMap[version].build_id);
+            window.location = initialPageData.reverse('diff', version);
```

Comment:
`version` now here can be also a part of a comment and not just number like `10 comment`?
would that be an issue here?

Reply:
No, `version` isn't the query itself, it's the value of the select, which is a build id. I should rename this variable.

Reply:
(updated)